### PR TITLE
State transfer fix data races

### PR DESCRIPTION
**Pull request by glevkovich**

### CODE REVIEW

File: tests/apollo/util/bft.py [hunk 1]
```diff
@@ -501,11 +501,15 @@ def start_replica(self, replica_id):
             if self.is_existing and self.config.stop_replica_cmd is not None:
                 self.procs[replica_id] = self._start_external_replica(replica_id)
             else:
+                my_env = os.environ.copy()
+                # Uncomment the next line to run Apollo tests with ThreadSanitizer
+                # my_env["TSAN_OPTIONS"] = "history_size=7 external_symbolizer_path=/usr/lib/llvm-6.0/bin/llvm-symbolizer"
```
Comment (HristoStaykov) [inline tests/apollo/util/bft.py:506]: Might be we can put this variable in apollo/CMakeLists.txt in the APOLLO_TEST_ENV?

**Review by HristoStaykov — APPROVED** [review]
LGTM! Thanks @glevkovich !

File: bftengine/src/bcstatetransfer/InMemoryDataStore.hpp [hunk 2]
```diff
@@ -163,7 +163,7 @@ class InMemoryDataStore : public DataStore {
 
   map<uint64_t, CheckpointDesc> descMap;
 
-  bool fetching = false;
+  std::atomic<bool> fetching = false;
```
Comment (dartdart26) [inline bftengine/src/bcstatetransfer/InMemoryDataStore.hpp:166]: Thanks for fixing that!

File: tests/apollo/util/bft.py (hunk 1 above)
Comment (glevkovich) [reply tests/apollo/util/bft.py]: As discussed, we moved this into APOLLO_TEST_ENV

**Pull request merged**